reports/GitalyFail/GitalyFail.tla

```diff
@@ -1,137 +1,117 @@
-\* Created Wed Jan 20 15:30:37 MSK 2021 by i00544730
+(* 
+    MIT License Copyright (c) 2021 CQFN
+    https://github.com/cqfn/degitx/blob/master/LICENSE
+    
+    DeGit Tech Report. 
+    Appendix 1. Gitaly fails explore.
+    
+    Situation:
+    -> 1.1 Preafect calls Gitally with data to commit.
+    |    2.1 Node process those data and call preafect with "prepared" state.
+    |    2.2 Preafect responses with "commit" or "abort" command, depends on quorum.
+    -- 1.2 Node writes data and response "acknowlege committed"
+    3.1 Preafect calls DB and increase generations.
+
+    Problem statement:
+    Node can write down data (make commit done) (step 1.2) 
+    and not response to Preafect, or Preafect can die on step 3.1.
+
+    Result:
+    At least ANY node where
+        Data ARE saved.
+        Generation is NOT incremented.
+    
+ *)
+------------------------------ MODULE GitalyFail ------------------------------
+EXTENDS TLC, Naturals, FiniteSets, Sequences
+
+CONSTANT RM
+
+VARIABLES rmState, rmVotedYes, rmVotedNo, rmAck, rmPrepared, rmCommitted, rmAborted, 
+    rmGenerations, targetGeneration
+
+\* Invariants
+TypeOK == \A r \in RM: rmState[r] \in [RM -> {"init", "working", "prepared", "committed", "aborted"}]
+
+Consistent == \A r \in RM : ~( \/ (/\ rmState[r] = "aborted"
+                                   /\ rmGenerations[r] = 1)
+                               \/ (/\ rmState[r] = "committed"
+                                   /\ rmGenerations[r] = 0)
+                              )
+
+-------------------------------------------------------------------------------
+========
+
+MasterNode = Head(RM)
+            
+GFInit ==   /\ rmState = [r \in RM |-> "init"]            
+            /\ rmGenerations = [r \in RM |-> 0]
+            /\ targetGeneration = 0
+            /\ rmVotedYes = {}
+            /\ rmVotedNo = {}
+            /\ rmAck = {}
+            /\ rmPrepared = {}
+            /\ rmCommitted = {}
+            /\ rmAborted = {}
+
+\* Step 1.1 Preafect call RM and swith it to working state
+RMRcvPack(r) == /\ rmState[r] = "init"
+                /\ rmState' = [rmState EXCEPT ![r] = "working"] \* All the Nodes are answered
+                
+\* Step 2.1 Node calls Preafect and send Vote
+NodeVote(r) == /\ r \notin rmVotedYes
+               /\ r \notin rmVotedNo
+               /\ rmState[r] = "working"
+               /\ (\/ (/\ rmState' = [rmState EXCEPT ![r] = "prepared"]
+                       /\ rmVotedYes' = rmVotedYes \union r
+                       /\ rmPrepared' = rmPrepared \union r) \* Prepared
+                   \/ (/\ rmState' = [rmState EXCEPT ![r] = "aborted"]
+                       /\ rmVotedNo' = rmVotedNo \union r
+                       /\ rmAborted' = rmAborted \union r)  \* Aborted
+                   )            
+
+\* Step 2.2 Preafect await for Quorum and send its decision
+isQuorumYes == /\ MasterNode \in rmVotedYes
+               /\ rmState[MasterNode] = "prepared" \* Doublecheck

```

Comment:
One of these conditions looks redundantly - it's not possible to get `MasterNode` in `rmVotedYes` and not get `rmState[MasterNode]` as `prepared`